module1-introduction-to-backend/r2.1-diving-into-rest-apis/README.md

```diff
@@ -0,0 +1,124 @@
+# Diving into REST APIs
+HTTP messages are how data is exchanged between a server and a client. There are two types of messages: requests sent by the client to trigger an action on the server, and responses, the answer from the server. The learning objectives of this lesson are:
+1. Understanding HTTP request types
+2. Understanding HTTP request and response structure
+3. Understanding HTTP response status codes
+
+## HTTP Requests
+In the previous lesson, we briefly talked about GET, POST, PUT and DELETE request methods.
+
+HTTP defines a set of request methods to indicate the desired action to be performed for a given resource. This set includes:
+- **GET**: Requests a representation of the specified resource. Requests using GET should only retrieve data.
+- **HEAD**: Asks for a response identical to that of a GET request, but without the response body.
+- **POST**: Used to submit an entity to the specified resource, often causing a change in state or side effects on the server.
+- **PUT**: Replaces all current representations of the target resource with the request payload.
+- **DELETE**: Requests to delete the specified resource.
+- **CONNECT**: Establishes a tunnel to the server identified by the target resource.
+- **OPTIONS**: Used to describe the communication options for the target resource.
+- **TRACE**: Performs a message loop-back test along the path to the target resource.
+- **PATCH**: Used to apply partial modifications to a resource.
+
+While developing APIs, GET and POST are the most commonly and frequently used methods.
```

Comment:
i would clarify that HEAD DELETE CONNECT OPTIONS TRACE and maybe even patch are almost never needed to be known